onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -1601,11 +1602,12 @@ including arg name, arg type (contains both type and shape).)pbdoc")
           },
           py::return_value_policy::reference_internal)
       .def_property_readonly(
-          "session_options", [](const PyInferenceSession* sess) -> const PySessionOptions& {
-            const auto& session_options = sess->GetSessionHandle()->GetSessionOptions();
-            return static_cast<const PySessionOptions&>(session_options);
+          "session_options", [](const PyInferenceSession* sess) -> const PySessionOptions* {
+            auto session_options = std::make_unique<PySessionOptions>();
+            session_options->value = sess->GetSessionHandle()->GetSessionOptions();
+            return session_options.release();
           },
-          py::return_value_policy::reference_internal)
+          py::return_value_policy::take_ownership)
```

Comment:
Note that this is no longer returning a const reference to the inference session's options. Instead, we need to return a heap-allocated `PySessionOptions` that contains a copy of the inference session's options.

`return_value_policy::take_ownership` tells python to manage the lifetime of the returned pointer.